grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -235,7 +239,8 @@ class StockMovementController {
         params.direction = (currentLocation == stockMovement.origin) ? StockMovementDirection.OUTBOUND :
                 (currentLocation == stockMovement.destination) ? StockMovementDirection.INBOUND : "ALL"
 
-        redirect(action: "list", params:params)
+
+        redirect(action: "list", params: params << ["flash": flash as JSON])
```

Comment:
I think this `params << ["flash": flash as JSON]` should happen in the `list` action while rendering the template.

Reply:
It cannot be in the `list` action, because there are two redirects:

- from remove action to list action
- from list action to react pages

Flash message is persisted only during one redirect